pandas/tests/groupby/test_categorical.py

```diff
@@ -420,6 +420,18 @@ def test_observed_groups(observed):
     tm.assert_dict_equal(result, expected)
 
 
+def test_observed_groups_with_nan(observed=True):
```

Comment:
You don't need `observed` as a keyword argument here; can just pass the literal value in the `groupby` call